src/utils/inline-styles.js

```diff
@@ -57,15 +59,18 @@ export default class InlineStyles extends LitElement {
       return;
     }
     const root = this._rootNode || this.ownerDocument;
-    const referenceCount = (referenceCounts.get(root) | 0) - 1;
+    const info = referenceCounts.get(root) || {};
+    const referenceCount = (info.referenceCount || 0) - 1;
     // remove or update reference-count metadata
     if (referenceCount < 1) {
       referenceCounts.delete(root); // give an early hint to GC
-      if (styleTagNode && styleTagNode.parentNode) {
-        styleTagNode.parentNode.removeChild(styleTagNode); // remove style node as well
+      const { style } = info;
+      if (style && style.parentNode) {
+        style.parentNode.removeChild(style); // remove style node in DOM
```

Comment:
This is still working, so this extra bugfix is ok. (Have testet my "old" bug with your new refactoring code.)